include/dali/core/math_util.h

```diff
@@ -192,6 +192,46 @@ float sinc(float x) {
   return std::sin(x) / x;
 }
 
+/// @brief Calculates power for integer arguments
+template <typename X, typename Y>
+DALI_HOST_DEV DALI_FORCEINLINE
+std::enable_if_t<std::is_integral<X>::value && std::is_unsigned<Y>::value,
+                  decltype(std::declval<X>() * std::declval<Y>())>
+ipow(X x, Y y) {
+  decltype(std::declval<X>() * std::declval<Y>()) acc = 1;
+  decltype(std::declval<X>() * std::declval<Y>()) pow_acc = x;
+  while (y > 0) {
+    if (y & 1) {
+      acc *= pow_acc;
+    }
+    pow_acc *= pow_acc;
+    y = y >> 1;
+  }
+  return acc;
+}
+
+/// @brief Calculates power for integer arguments
+template <typename X, typename Y>
+DALI_HOST_DEV DALI_FORCEINLINE
+std::enable_if_t<
+    std::is_integral<X>::value && std::is_integral<Y>::value && std::is_signed<Y>::value,
+    decltype(std::declval<X>() * std::declval<Y>())>
```

Comment:
I have my doubts regarding this promotion - if anything, I'd say it should rather be
```suggestion
    decltype(std::declval<X>() << std::declval<Y>())>
```